app/views/contexts/summary/group.py

```diff
@@ -168,6 +171,38 @@ def _build_blocks_and_links(
 
                     self.placeholder_text = list_summary_element["empty_list_text"]
 
+            elif block["type"] == "ListCollectorContent":
+                list_collector_content_section: ImmutableDict | None = (
```

Comment:
This is identical to the "ListCollector" case except for the `if view_submitted_response` block, I think it could be collapsed into one with that if block only applying for a ListCollector

Reply:
Well, there is one other difference, it initialises different class, but I tried to make it work and reduce duplication.